nimbus-ui/nimbusui/src/app/components/platform/grid/table.component.ts

```diff
@@ -570,6 +570,8 @@ export class DataTable extends BaseTableElement implements ControlValueAccessor
     filterCallBack(e: any) {
         this.totalRecords = e.filteredValue.length;
         this.updatePageDetailsState();
+        this.selectedRows.length = 0
+        this.dt.toggleRowsWithCheckbox(e, true);
```

Comment:
This is done because just setting selectedRows property to 0 was not updating the css on table (the row would still show up as selected, however when submitting it was not including the row). this call updates the css as well.